### ISSUE TITLE
Column attribute doesn't function with vertically merged cells

### ISSUE THREAD

**Issue by danyill** [asciidoctor/asciidoctor#2521]
The following table does not render correctly:

```
[#fishwhiskers,reftext='{table-caption} {counter:tabnum}']
.Fishwhiskers
[cols="1*^.^1,1*^.^1m,1*^.^3,1*^.^5",options="header,unbreakable"]
|===
|*a* |*b* |*c* |*d*
.4+|a1 |b1 |c1 |d1

|b2 |c2 |d2
|b3 |c3 |d3
|b4 |c4 |d4

.5+|a5 |b5 |c5 |d5
|b6 |c6 |d6
|b7 |c7 |d7
|b8 |c8 |d8
|b9 |c9 |d9

|===
```

The monsopace attribute appears to jump columns, so it applies to:

b1,c2,c3,c4,b5,c6,c7,c8,c9

It should only apply (unless I err) to

b1,b2,b3,b4,b5,b6,b7,b8,b9

Screenshot below.

![image](https://user-images.githubusercontent.com/674783/34338369-f388f318-e9cc-11e7-9a42-ede157326f55.png)

Still loving Asciidoctor and wishing you a very Merry Christmas!

**Comment by danyill** [comment]
I should add:
```
⋊> ~ asciidoctor -v                                                                                                                  23:42:08
Asciidoctor 1.5.6.1 [http://asciidoctor.org]
Runtime Environment (ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]) (lc:UTF-8 fs:UTF-8 in:- ex:UTF-8)
⋊> ~   
```

**Comment by danyill** [comment]
A simpler table to reproduce this is:

```
:nofooter:
:linkcss:

[cols="1m,1s,1",options="header"]
|===
|*a* |*b* |*c*
.3+|a1 |b1 |c1

|b2 |c2
|b3 |c3
|===
```
Effectively the vertical span is not taken into account -- we see the monospace and the bold column attributes offset.

![image](https://user-images.githubusercontent.com/674783/34345740-1f1bff0a-ea56-11e7-956e-0d0883f0dd59.png)

I had a somewhat unsuccessful attempt to debug this. 

For th cells b2 and b3 the `colnumber` is `1` when I think it should be `2`

Basically where there is a vertical span, the column number (`colnumber`) on successive rows is not adjusted. I'll do a PR if I can figure it out...
